Assets/Scripts/Game/Core/UniqueID.cs

```diff
@@ -0,0 +1,55 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace RPG
+{
+    [RequireComponent(typeof(Entity))]
+    public class UniqueID : MonoBehaviour
+    {
+        [SerializeField] private string uniqueID;
+
+        private static Dictionary<string, GameObject> uniqueObjects = new Dictionary<string, GameObject>();
+
+        protected void OnValidate()
```

Comment:
You'll need to add the UID to the static dictionary otherwise this won't do anything.